### PR TITLE
refactor(client_core): :recycle: Port FFE to Rust and remove all C++ code

### PR DESCRIPTION
**Pull request by zmerp**

### CODE REVIEW

File: alvr/client_core/src/graphics/stream.rs [hunk 1]
```diff
@@ -41,240 +38,276 @@ impl StreamRenderer {
...
         }
+
+        self.context.queue.submit(iter::once(encoder.finish()));
     }
 }
+
+pub fn foveated_encoding_shader_constants(
+    expanded_view_resolution: UVec2,
+    config: FoveatedEncodingConfig,
+) -> (UVec2, HashMap<String, f64>) {
```
Comment (The-personified-devil) [inline alvr/client_core/src/graphics/stream.rs:238]: I only roughly checked all of this, it's just gonna have to be right

**Review by The-personified-devil — COMMENTED** [review]
LGTM, check the comments tho
Also the diff is atrocious so the review might be a bit not so thorough

**Review by The-personified-devil — APPROVED** [review]

File: alvr/client_core/build.rs [hunk 2]
```diff
@@ -1,59 +1,13 @@
...
-                .unwrap();
-        }
-
         println!("cargo:rustc-link-lib=log");
         println!("cargo:rustc-link-lib=EGL");
         println!("cargo:rustc-link-lib=GLESv3");
         println!("cargo:rustc-link-lib=android");
 
         #[cfg(feature = "link-stdcpp-shared")]
         println!("cargo:rustc-link-lib=c++_shared");
```
[the remark this answers is not in the document]
Comment (zmerp) [reply alvr/client_core/build.rs:11]: This is needed for PhoneVR

File: alvr/client_core/src/graphics/stream.rs (hunk 1 above)
Comment (zmerp) [reply alvr/client_core/src/graphics/stream.rs:238]: Yes, I've tested all settings combinations, works perfectly like before

**Pull request merged**